File: web/js/physics.js
class Physics {
    applyGravity(rocket, planets, dt) {
        for (const planet of planets) {
            const dx = rocket.x - planet.x;
            const dy = rocket.y - planet.y;
            const distSq = dx * dx + dy * dy;
            const dist = Math.sqrt(distSq);
            
            if (dist > 0) {
                const gravityForce = CONFIG.GRAVITY_CONSTANT * planet.radius / distSq;
                const ax = -gravityForce * dx / dist;
                const ay = -gravityForce * dy / dist;
                
                rocket.vx += ax * dt;
                rocket.vy += ay * dt;
            }
        }
    }

    checkCollision(rocket, planet) {
        const dist = rocket.distanceTo(planet);
        return dist < planet.radius + rocket.radius;
    }
}

File: src/systems/physics.py
"""
Физическая система - гравитация, движение, столкновения
"""

import math
from src.config import Config


class PhysicsSystem:
    def __init__(self):
        self.gravity_constant = Config.GRAVITY_CONSTANT
        
    def apply_gravity(self, rocket, planets):
        """Применить гравитацию планет к ракете"""
        for planet in planets:
            dx = rocket.x - planet.x
            dy = rocket.y - planet.y
            dist_sq = dx**2 + dy**2
            dist = math.sqrt(dist_sq)
            
            if dist > 0:
                # Сила гравитации (упрощённая)
                gravity_force = self.gravity_constant * planet.radius / dist_sq
                
                # Направление к планете
                ax = -gravity_force * dx / dist
                ay = -gravity_force * dy / dist
                
                rocket.vx += ax
                rocket.vy += ay
